File: cts.core.api.tourplanner/cts.core.svc.contracts/Tours/TourDto.cs
using cts.core.svc.contracts.TourLogs;
using cts.core.svc.domain.OpenRoute;

namespace cts.core.svc.contracts.Tours;

public record TourDto(
    Guid TourGuid,
    Guid UserGuid,
    string Name,
    string Description,
    string From,
    string To,
    string TransportName,
    double TourDistanceInMeters,
    int EstimatedTimeMinutes,
    int Rating,
    List<TourLogDto> TourLogs,
    RouteGeometry? RouteGeometry
);

File: cts.core.ui.tourplanner/frontend/src/app/contracts/TourDto.ts
import { TourLogDto } from './TourLogDto';

export interface TourDto {
  tourGuid: string;
  userGuid: string;
  name: string;
  description: string;
  from: string;
  to: string;
  transportName: string;
  tourDistanceInMeters: number;
  estimatedTimeMinutes: number;
  rating: number;
  tourLogs: TourLogDto[];
}
